### PR TITLE
Use rbenv for latest Ruby version, not S3 file

### PR DESCRIPTION
**Pull request by croaky**

### CODE REVIEW

**Comment by geoffharcourt** [comment]
Given that ruby-build/rbenv will be installed at the point where we'll be attempting to determine what version to install, I like this approach. If the `grep`/`sed` maneuvering here feels awkward, we could also try leveraging [ruby-versions](https://github.com/postmodern/ruby-versions/blob/master/ruby/versions.txt), which has an MRI-only file that we could just pull the last line from.

**Comment by croaky** [comment]
@geoffharcourt That could be interesting. I just made that change to try it out. @thoughtbot/shell any opinions on which is more portable?

We're already using `grep` in the script, but `sed` would have been a new introduction.

File: mac [hunk 1]
```diff
@@ -157,7 +157,14 @@ brew_install_or_upgrade 'openssl'
 brew unlink openssl && brew link openssl --force
 brew_install_or_upgrade 'libyaml'
 
-ruby_version="$(curl -sSL http://ruby.thoughtbot.com/latest)"
+find_latest_ruby() {
+  curl -sSL \
+    'https://raw.githubusercontent.com/postmodern/ruby-versions/master/ruby/versions.txt' \
+    | tail -n 2 \
```
Comment (gabebw) [inline mac]: It looks like this is grabbing the second-to-last version. Why not grab the latest one (with `tail -n1`)?

File: mac [hunk 2]
```diff
@@ -157,7 +157,14 @@ brew_install_or_upgrade 'openssl'
 brew unlink openssl && brew link openssl --force
 brew_install_or_upgrade 'libyaml'
 
-ruby_version="$(curl -sSL http://ruby.thoughtbot.com/latest)"
+find_latest_ruby() {
+  curl -sSL \
+    'https://raw.githubusercontent.com/postmodern/ruby-versions/master/ruby/versions.txt' \
```
Comment (gabebw) [inline mac]: I don't think these quotes are necessary.

**Comment by gabebw** [comment]
If we do use sed, I'd use ` `  instead of `[[:space:]]`, because I think ` ` is clearer.

**Comment by gabebw** [comment]
After thinking about it for a while, I think I prefer `rbenv install -l`:

* We know it's up to date when we're running it, so it should be just as current as the versions file from github
* It's faster to run locally than to make the request to GitHub

File: mac (hunk 1 above)
Comment (croaky) [inline mac]: There was an empty line at the end of the file.

Comment (mike-burns) [inline mac]: I don't see that:

```sh
~% curl -sSL https://raw.githubusercontent.com/postmodern/ruby-versions/master/ruby/versions.txt | tail -1
2.3.0
~% curl -sSL https://raw.githubusercontent.com/postmodern/ruby-versions/master/ruby/versions.txt | tail -2 | head -1
2.3.0-preview2
```

Let's watch for that by deleting blank lines:

```sh
curl -sSL https://raw.githubusercontent.com/postmodern/ruby-versions/master/ruby/versions.txt | sed '/^ *$/d' | tail -n 1
```

**Comment by jferris** [comment]
I agree with @gabebw. Can we let rbenv figure this out for us?

File: mac (hunk 1 above)
Comment (thorncp) [inline mac]: Do we know that a stable version will always be the last listing in that file? Will we start auto installing `2.4.0-preview1` (or whatever the next version is) once it's available? Is that a problem?

**Comment by croaky** [comment]
Reverted back to `rbenv -l`

**Comment by croaky** [comment]
Any thoughts on the portability of `sed`? It's a new dependency to Laptop.

**Comment by mike-burns** [comment]
http://pubs.opengroup.org/onlinepubs/9699919799/utilities/sed.html

sed is portable, the `-e` flag is portable, the `s` command is portable, and the regexp you used is portable.

File: mac [hunk 3]
```diff
@@ -157,7 +157,11 @@ brew_install_or_upgrade 'openssl'
 brew unlink openssl && brew link openssl --force
 brew_install_or_upgrade 'libyaml'
 
-ruby_version="$(curl -sSL http://ruby.thoughtbot.com/latest)"
+find_latest_ruby() {
+  rbenv install -l | grep -v - | tail -1 | sed -e 's/^ *//'
```
Comment (gabebw) [inline mac]: > Do we know that a stable version will always be the last listing in that file? Will we start auto installing 2.4.0-preview1 (or whatever the next version is) once it's available? Is that a problem?

@thorncp The `grep -v -` should take care of that.

Comment (thorncp) [inline mac]: :+1: wicked

**Comment by gabebw** [comment]
CI is failing because `apt-get update` is failing -- maybe we need to point it at a different repository [on this line](https://github.com/thoughtbot/laptop/blob/73f353ef39dd6922b2937656935268c77def95be/.travis.yml#L4)?

**Comment by croaky** [comment]
Thanks, @gabebw https://github.com/thoughtbot/laptop/pull/431 for CI.

**Pull request merged**